aggregator/src/app/HealthService.ts

```diff
@@ -0,0 +1,192 @@
+import { axiod } from "../../deps.ts";
+import BundleTable from "./BundleTable.ts";
+import AppEvent from "./AppEvent.ts";
+import assert from "../helpers/assert.ts";
+import * as env from "../env.ts";
+
+abstract class HealthIndicator {
+  constructor(
+    public emit: (evt: AppEvent) => void,
+  ){}
+  abstract name: string;
+  status: ResourceHealth = ResourceHealth.Unhealthy;
+  details: string | undefined;
+  abstract checkHealth(): Promise<void> | void;
+}
+
+export enum ResourceHealth {
+  Healthy = 'HEALTHY',
+  Unhealthy = 'UNHEALTHY'
+}
+
+export class AggregatorServiceHealthCheck extends HealthIndicator {
+  // Starts out in the Unhealthy state by default until it can be verified as Healthy
+  name = 'Aggregator';
+  status: ResourceHealth = ResourceHealth.Unhealthy;
+  
+  async checkHealth(): Promise<void> {
+    try {
+      const response = await axiod.get(new URL(env.ORIGIN, "/Bundle/health").toString());
+      if (response.status === 200) {
+        this.status = ResourceHealth.Healthy;
+        this.emit({
+          type: "service-healthy",
+          data: {
+            name: this.name,
+            status: ResourceHealth.Healthy,
+          },
+        });
+      } else {
+        this.status = ResourceHealth.Unhealthy;
+        this.details = `Aggregator returned status code ${response.status}`;
+        this.emit({
+          type: "service-unhealthy",
+          data: {
+            name: this.name,
+            status: ResourceHealth.Unhealthy,
+            detail: `Aggregator returned status code ${response.status}`,
+          },
+        });
+      }
+    } catch (e) {
+      this.status = ResourceHealth.Unhealthy;
+      this.details = e.message;
+      this.emit({
+        type: "service-unhealthy",
+        data: {
+          name: this.name,
+          status: ResourceHealth.Unhealthy,
+          detail: e.message,
+        },
+      });
+      //console.log(`HEALTH: ${this.name} is unhealthy.`, e.message);
+    }
+  }
+}
+
+export class RPCServiceHealthCheck extends HealthIndicator  {
+  // Starts out in the Unhealthy state by default until it can be verified as Healthy
+  name = 'RPC';
+  status: ResourceHealth = ResourceHealth.Unhealthy;
+
+  async checkHealth(): Promise<void> {
+    try {
+      const response = await axiod.get(env.RPC_URL);
```

Comment:
Good idea checking the RPC. We should probably send them a concrete request though, I'm not sure whether all providers will respond nicely to a blank request.

You can get chainId by using this for the body:

```json
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "eth_chainId",
  "params": []
}
```